js_modules/dagit/packages/core/src/nav/__tests__/useRepositoryLocationReload.test.tsx

```diff
@@ -171,15 +177,13 @@ describe('useRepositoryReloadLocation', () => {
     });
 
     // Set the location entry to `LOADED`, which should terminate polling.
-    rerender(
-      <TestProvider apolloProps={{mocks: defaultMocks}}>
-        <Test />
-      </TestProvider>,
-    );
-
-    jest.runAllTicks();
-
     await waitFor(() => {
+      rerender(
+        <TestProvider apolloProps={{mocks: defaultMocks}}>
+          <Test />
+        </TestProvider>,
+      );
+      jest.runAllTicks();
```

Comment:
if the re-render is doing async things then we should wrap it in `act` rather than `waitFor` imo.

Reply:
Ahh we can't do this because for whatever reason, `rerender` does not return a promise but IS asynchornous, so you can't `await act(async () => await rerender())`